addons/bus/static/src/js/crosstab_bus.js

```diff
@@ -0,0 +1,355 @@
+        $(window).off('unload.' + this._id);
+    },
+    //--------------------------------------------------------------------------
+    // Public
+    //--------------------------------------------------------------------------
+    /**
+     * Share the bus channels with the others tab by the local storage
+     *
+     * @override
+     */
+    addChannel: function () {
+        this._super.apply(this, arguments);
+        this._callLocalStorage('setItem', 'channels', this._channels);
+    },
+    /**
+     * Share the bus channels with the others tab by the local storage
+     *
+     * @override
+     */
+    deleteChannel: function () {
+        this._super.apply(this, arguments);
+        this._callLocalStorage('setItem', 'channels', this._channels);
+    },
+    /**
+     * Tells whether this bus is related to the master tab.
+     *
+     * @returns {boolean}
+     */
+    isMasterTab: function () {
+        return this._isMasterTab;
+    },
+    /**
+     * Use the local storage to share the long polling from the master tab.
+     *
+     * @override
+     */
+    startPolling: function () {
+        var self = this;
+        if (!this._isRegistered) {
+            this._isRegistered = true;
+
+            var peers = this._callLocalStorage('getItem', 'peers', {});
+            peers[this._id] = new Date().getTime();
+            this._callLocalStorage('setItem', 'peers', peers);
+
+            $(window).on('unload.' + this._id, this._onUnload.bind(this));
+
+            if (!this._callLocalStorage('getItem', 'peers')) {
+                this._startElection();
+            }
+
+            this._heartbeat();
+
+            if (this._isMasterTab) {
+                this._callLocalStorage('setItem', 'channels', this._channels);
+                this._callLocalStorage('setItem', 'options', this._options);
+            } else {
+                this._channels = this._callLocalStorage('getItem', 'channels', this._channels);
+                this._options = this._callLocalStorage('getItem', 'options', this._options);
+            }
+            return;  // startPolling will be called again on tab registration
+        }
+
+        if (this._isMasterTab) {
+            this._super.apply(this, arguments);
+        }
+    },
+    /**
+     * Share the option with the local storage
+     *
+     * @override
+     */
+    updateOption: function () {
+        this._super.apply(this, arguments);
+        this._callLocalStorage('setItem', 'options', this._options);
+    },
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+    /**
+     * Call local_storage service
+     *
+     * @private
+     * @param {string} method (getItem, setItem, removeItem, on)
+     * @param {string} key
+     * @param {any} param
+     * @returns service information
+     */
+    _callLocalStorage: function (method, key, param) {
```

Comment:
I am not fond of the `_callLocalStorage` function, because it works with `getItem` and `setItem` but not with `onStorage`.